rfcs/0132-meson-builds-nix.md

```diff
@@ -0,0 +1,127 @@
+[drawbacks]: #drawbacks
+
+Some possible drawbacks:
+
+- A new build system would require changes on the code
+  + On the other hand, such changes are likely to improve the code base.
+  
+- A new build system requires the developers become familiarized with it
+  - Specially when this build system uses its own description language.
+  + However, Meson is well documented, and its Python-esque language is easy to grasp.
+
+- A new build system indirectly brings its own dependencies to the Nix project
+  - In particular, the reference implementation of Meson is written in Python.
+  - Further, this reference implementation generates script files meant to be consumed by Ninja.
+  - Ninja is a tool written in C++ that acts like a `make` replacement.
+  - This particular setting brings concerns about complexifying the bootstrap route.
+  + Given that Nix is currently written in C++, we can assume a C++ compiler as part of such a bootstrap route.
+  + There are full-featured alternative tools that replace Meson and Ninja. 
+    + Namely, Muon and Samurai are implementations of Meson and Ninja that require only a C compiler and a set of POSIX standard tools.
+
+- A new build system would require new strategies from the end users
+  - In particular, package managers that deploy Nix for their respective platforms.
+  + However, Meson and Ninja are a widespread toolset.
+    + Many open source projects use Meson, from mpv and dosbox-staging to Xorg and GNOME
+    + According to Repology, Meson is present in 53 package manager's families
+
+- The transition between between the old and new build systems should be smooth
+  - A wrapper script, maybe?
+  + Meson is not an obscure project; a mere documentation update should suffice
+
+# Alternatives
+[alternatives]: #alternatives
+
+The alternatives are
+
+- Doing nothing
+
+  It would keep the current code confusing and harder to work with.
+  
+- Other building systems (cmake, waf, premake etc.)
+  - Their strenghts and weaknesses should be evaluated.
+    - Tools strongly tied to other programming languages are strongly discouraged, because they further complexifies the bootstrap route as discussed above.
+      - Namely, waf is basically a Python library, whereas premake is a Lua library.
+    - Cmake has many noteworthy advantages:
+      + Can generates Make- and Ninja-compatible scripts;
+      + Supports Windows NT and MacOS platforms
+      + Supports typical high level idiomatic constructions;
+      - On the other hand, cmake language is arguably more complex.
+      - Both Meson and Cmake support Apple Xcode and Microsoft MSVC project file formats
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+Questions that deserve further inquiry:
+
+- Unexpected interactions with Meson and Ninja
+  - Specially, vendoring and reproducibility.
+
+# Future work
+[future]: #future-work
+
+- Deprecate and remove the current quasi-autotools scripts
+- Backport the new build system to Nix 2.3
```

Comment:
```suggestion
```

We definitely shouldn't be backporting the build system, that's far out of scope for maintenance branches.

Reply:
@p01arst0rm This is a political minefield I would just removed it. I personally like the non-flakes improvements  in newer Nix enough to not miss 2.3, and any of the 2.3 holdouts are free to maintain a non-official backport.

Reply:
Removed.